src/frontend/src/flows/recovery/setupRecovery.ts

```diff
@@ -131,40 +134,62 @@ export const setupPhrase = async (
   userNumber: bigint,
   connection: AuthenticatedConnection
 ): Promise<"ok" | "error" | "canceled"> => {
-  const name = "Recovery phrase";
+  const res = await phraseWizard({
+    userNumber,
+    operation: "create",
+    uploadPhrase: (pubkey) =>
+      withLoader(() =>
+        connection.add(
+          "Recovery phrase",
+          { seed_phrase: null },
+          { recovery: null },
+          pubkey,
+          { unprotected: null }
+        )
+      ),
+  });
+
+  if (typeof res === "object" && "ok" in res) {
+    return "ok";
+  } else if (typeof res === "object" && "error" in res) {
+    return "error";
+  } else {
+    return res;
+  }
+};
+
+// Set up a recovery phrase
+export const phraseWizard = async ({
+  userNumber,
+  operation,
+  uploadPhrase,
+}: {
+  userNumber: bigint;
+  operation: "create" | "reset";
+  uploadPhrase: (pubkey: DerEncodedPublicKey) => Promise<void>;
+}): Promise<{ ok: SignIdentity } | { error: unknown } | "canceled"> => {
```

Comment:
any particular reason not using `canceled` as one of the variant?

that would spare these `typeof res === "object"` checks

```
Promise<{ ok: SignIdentity } | { error: unknown } | {canceled: never}>
```

P.S.: not sure about the `never` above

Reply:
Not strong on my side but removing `typeof res === "object"` would have ease my reading